pydantic/types.py

```diff
@@ -358,6 +361,12 @@ def validate(self, value: UUID, _: core_schema.ValidationInfo) -> UUID:
     def __hash__(self) -> int:
         return hash(type(self.uuid_version))
 
+    def __iter__(self) -> Iterator[object]:  # type: ignore
+        if 'hypothesis' in sys.modules:
+            from hypothesis import strategies as st
+
+            yield st.uuids(version=self.uuid_version)
```

Comment:
This is not actually an iterator, just an iterable, so the type-check error is actually correct.  Try `return iter([st.uuids(version=self.uuid_version)])` or `return iter([])` in the different branches?

You could also annotate `-> Iterator[st.SearchStrategy[UUID]]` with the import in `if TYPE_CHECKING:`, to give users more-precise type annotations.  I think there's enough indirection that this won't actually help in practice though.